pkg/extender/extender.go

```diff
@@ -219,8 +236,7 @@ func (e *Extender) getNodeScore(
 						for _, rack := range rackInfo.PreferredLocality {
 							if rack == nodeRack || nodeRack == "" {
 								for _, datanode := range volumeInfo.DataNodes {
-									if e.isHostnameMatch(idMap[datanode], address.Address) ||
-										e.isHostnameMatch(idMap[datanode], node.Name) {
+									if e.isNodeMatch(&node, idMap[datanode]) {
```

Comment:
this can panic in `isNodeMatch` if datanode is not present in idMap.

Reply:
Ignoring nil node in isNodeMatch